microsoft/testsuites/osu/osusuite.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from pathlib import Path
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.features import Gpu, GpuEnabled, SerialConsole
+from lisa.operating_system import Posix
+from lisa.tools import Make, Tar, Wget
+
+OSU_MPI_LOCATION = (
+    "https://mvapich.cse.ohio-state.edu/download/mvapich/mv2/mvapich2-2.3.7-1.tar.gz"
+)
+OSU_MPI_FILE_NAME = "mvapich2.tgz"
+
+OSU_BENCH_LOCATION = (
+    "http://mvapich.cse.ohio-state.edu/download/mvapich/osu-micro-benchmarks-6.2.tar.gz"
+)
+OSU_BENCH_FILE_NAME = "osu-bench.tgz"
+
+
+@TestSuiteMetadata(
+    area="hpc",
+    category="functional",
+    name="OSU Bench",
+    description="""
+    This test suite runs the OSU Micro-Benchmarks MPI test cases.
+    """,
+)
+class OSUTestSuite(TestSuite):
+    TIMEOUT = 200000
+
+    @TestCaseMetadata(
+        description="""
+            This test case runs GPU/CPU MPI latency.
+
+            Steps:
+            1. Install MVAPICH;
+            2. Install OSU Micro-Benchmarks;
+            3. Run GPU/CPU collective/latency tests in a single node.
+
+        """,
+        timeout=TIMEOUT,
+        requirement=simple_requirement(
+            supported_features=[GpuEnabled(), SerialConsole],
+        ),
+        priority=1,
+    )
+    def run_osu_bench(self, node: Node, log: Logger, log_path: Path) -> None:
+
+        _install_osu_mpi(node)
+        in_path = _install_osu_bench(node)
+
+        gpu_count = node.capability.gpu_count
+        gpu = node.features[Gpu]
+        expected_count = gpu.get_gpu_count_with_lspci()
+        assert_that(gpu_count).described_as(
+            "Expected device count didn't match actual device count from lspci!"
+        ).is_equal_to(expected_count)
+
+        out_path = node.get_working_path() / "osu-micro-benchmarks-6.2"
+        tests = [
+            "allgather",
+            "allreduce",
+            "alltoall",
+            "bcast",
+            "gather",
+            "reduce_scatter",
+            "reduce",
+            "scatter",
+        ]
+        for test in tests:
+            node.execute(
+                f"mpirun -np {gpu_count} python3 {in_path}/python/run.py"
+                f" --benchmark {test} --min 128 --max 10000000 >> "
```

Comment:
when I tried this command with gpu_count = 1, it said.
```
This test requires at least two processes
```
I checked the output of `mpirun -h`, so it should be cpu count.
```
-n/-np {value}                   number of processes
```